core/services/bill.py

```diff
@@ -115,4 +118,25 @@ def cancelDeposit(self,id) -> None:
 
     def recordPayment(self,id):
         self.bill_repo.updatePayment(id,{"is_outstanding":False})
+        return 
+    
+    def getAllPaymentType(self) -> List[PaymentType]:
+        return self.bill_repo.listPaymentType()
+    
+    def getPaymentTypeById(self,id:int) -> PaymentType:
+        return self.bill_repo.getPaymentTypeById(id)
+    
+    def createPaymentTYpe(self,paymentType) -> None:
+        self.bill_repo.persistPaymentType({"name":paymentType.name,"is_default":paymentType.is_default})
```

Comment:
If you make the changes requested for payment type persist and update function. You can directly pass the paymentType.